fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistentStorageUtils.java

```diff
@@ -74,26 +79,52 @@ private OCFLPersistentStorageUtils() {
     private static final String FEDORA_METADATA_SUFFIX = "/" + FedoraTypes.FCR_METADATA;
 
     /**
-     * Returns the relative subpath of the resourceId based on the ancestor's resource id.
+     * Returns the relative subpath of the resourceId based on the root object's resource id.
      *
-     * @param ancestorResourceId The ancestor resource
-     * @param resourceId         The identifier of the resource whose subpath you wish to resolve.
+     * @param rootObjectId The fedora root object identifier
+     * @param resourceId   The identifier of the resource whose subpath you wish to resolve.
      * @return The resolved subpath
      */
-    public static String relativizeSubpath(final String ancestorResourceId, final String resourceId) {
-        if (resourceId.equals(ancestorResourceId)) {
-            return resourceId;
-        } else if (resourceId.startsWith(ancestorResourceId)) {
-            return resourceId.substring(ancestorResourceId.length() + 1);
+    public static String relativizeSubpath(final String rootObjectId, final String resourceId) {
+        if (resourceId.equals(rootObjectId)) {
+            return resourceId.substring(FEDORA_ID_PREFIX.length());
+        } else if (resourceId.startsWith(rootObjectId)) {
+
+            final var rawSubpath = resourceId.substring(rootObjectId.length() + 1);
+            //grab the last path segment of the rootObjectId
+            final var lastPathSegment = rootObjectId.substring(rootObjectId.lastIndexOf("/") + 1);
+
+            if (rawSubpath.endsWith(FCR_ACL) || rawSubpath.endsWith(FCR_METADATA)) {
+                final var lastSlashIndex = rawSubpath.indexOf("/");
+                String beforeSlash = null;
+                String suffix = rawSubpath;
+
+                if (lastSlashIndex > -1) {
+                    beforeSlash = rawSubpath.substring(0, lastSlashIndex);
+                    suffix = rawSubpath.substring(lastSlashIndex + 1);
+                }
+
+                String translatedSuffix = null;
+                if (suffix.equals(FCR_METADATA)) {
+                    translatedSuffix = "-description";
+                } else {
+                    translatedSuffix = "-acl";
+                }
+
+                return lastPathSegment + (beforeSlash != null ? beforeSlash : "") + translatedSuffix;
```

Comment:
You don't add a slash between `lastPathSegment` and `beforeSlash` here but you add one between `lastPathSegment` and `rawSubpath` below?

Reply:
This has been addressed.